src/shipping/strategies/amazon/amazon-pay-shipping-strategy.spec.ts

```diff
@@ -269,4 +269,22 @@ describe('AmazonPayShippingStrategy', () => {
         expect(store.dispatch).toHaveBeenCalledWith(action);
         expect(output).toEqual(store.getState());
     });
+
+    it('updates consignment shipping address with custom fields for amazon pay on strategy address update', async () => {
+        const strategy = new AmazonPayShippingStrategy(store, consignmentActionCreator, paymentMethodActionCreator, remoteCheckoutActionCreator, scriptLoader);
+        const options = {};
+        const address = getShippingAddressWithCustomFields();
+        const action = of(createAction(ConsignmentActionType.UpdateConsignmentRequested));
+
+        jest.spyOn(consignmentActionCreator, 'updateAddress')
+            .mockReturnValue(action);
+
+        jest.spyOn(store, 'dispatch');
+
+        const output = await strategy.updateAddress(address, options);
```

Comment:
might be worth ensuring that `store.getState().shippingAddress.getShippingAddress()` returns an address different than the one in `getShippingAddressWithCustomFields`, to make sure the merging of the data is done correctly